### PR TITLE
Support block & unstable statuses

### PR DESCRIPTION
**Pull request by d2lam**
Support `BLOCKED` and `UNSTABLE` status. 
Previously, we only allowed update `RUNNING` or `QUEUED`, but we want to allow updating `BLOCKED` and `UNSTABLE`. 
They should skip on updating `meta` and `endTime`.
`BLOCKED` -> `SUCCESS` is ok, but `UNSTABLE` is not allowed to be updated to something else. 

Related:
https://github.com/screwdriver-cd/screwdriver/issues/653
https://github.com/screwdriver-cd/screwdriver/issues/1122

### CODE REVIEW

File: plugins/builds/update.js [hunk 1]
```diff
@@ -96,8 +104,9 @@ module.exports = () => ({
                         buildLink:
                             `${buildFactory.uiUri}/pipelines/${pipeline.id}/builds/${id}`
                     });
-                    // Guard against triggering non-successful builds
-                    if (desiredStatus !== 'SUCCESS') {
+
+                    // Guard against triggering non-successful or unstable builds
+                    if (desiredStatus !== 'SUCCESS' || build.status === 'UNSTABLE') {
```
Comment (jithin1987) [inline plugins/builds/update.js]: won't `if (build.status !== 'SUCCESS')` do ?

Comment (d2lam) [reply plugins/builds/update.js]: `build.status` is current status. `desiredStatus` is the new status

We only want to trigger if the new status is `SUCCESS`, and the current status is not `UNSTABLE`.

**Comment by d2lam** [comment]
@tkyi  I implemented it based on the acceptance criteria `If a build's status is Unstable don't change it to Success at end of build.` and it shouldn't trigger next build according to @jithin1987 

I think the intention is user can update a build to `UNSTABLE` if they think something is not right. For example, coverage is too low. Therefore, it should probably not trigger the next build, and should not be reupdated to `SUCCESS` at the end of build.

File: plugins/builds/update.js (hunk 1 above)
Comment (d2lam) [reply plugins/builds/update.js]: Oh I see at this point `build.status` should be the new status already. We can use that check instead like you said.

**Comment by jithin1987** [comment]
> I think I'm still confused what UNSTABLE is used for. Is it just an arbitrary label put on a build instead of SUCCESS or FAILURE? Should UNSTABLE builds still trigger the next build?

UNSTABLE builds should not trigger next builds. It's a build status which build can call to report itself as UNSTABLE, we don't control the logic on what makes a build UNSTABLE.

File: plugins/builds/update.js [hunk 2]
```diff
@@ -74,14 +75,21 @@ module.exports = () => ({
...
 
-                    // Everyone is able to update the status
-                    build.status = desiredStatus;
-                    if (statusMessage) {
-                        build.statusMessage = statusMessage;
+                    // UNSTABLE -> SUCCESS needs to update meta and endtime.
+                    // However, the status itself cannot be updated to SUCCESS
+                    if (build.status !== 'UNSTABLE') {
+                        build.status = desiredStatus;
+                        if (statusMessage) {
```
Comment (jithin1987) [inline plugins/builds/update.js]: should `statusMessage` be restricted for UNSTABLE ?

Comment (d2lam) [reply plugins/builds/update.js]: I don't think so. Moved it out 👍

**Pull request merged**